### DISCUSSION TITLE
Can I use this with Node?

### DISCUSSION THREAD

**Discussion by bpetty-interlace** [FGasper/zmodemjs#32] [Q&A]
I am working on a Node based BBS I am putting together.
I would like to use ZModem to transfer files, but could not get passed the documentation challenges.
Can I use this library to automatically initiate a file transfer to my SyncTerm client?
I assume I would need to pass it my net socket object.  Any advice @FGasper ?

**Comment by FGasper** [comment]
Have a look at the unit tests in the repo for node.js usage examples.

**Reply by bpetty-interlace** [reply]
Thank you for the quick reply @FGasper.  I looked over the tests, but still seem to be lost.
I was asking specifically about how one would use this with node's Net TCP Server socket.

For example, importing 'net' and creating a 'Net.Server()' object.
Then waiting for the connection: server.on('connection', function(socket) { ...

I was expecting to somehow pass zmodemjs this socket object and simply call something like: zmod.send(socket, buffer);
Where buffer is a Buffer object I populated doing a fs read.

Looking at the api design, there is obviously a lot more going on than the simple .send(...) interface I just described.  Would I be correct in assuming that I would need to write a wrapper for zmodemjs to get it working with a socket?

**Reply by FGasper** [reply]
@bpetty-interlace Hey sorry, I lost track of this thread.

Did you ever sort this out?

**Reply by brandonmpetty** [reply]
No, I never did.  Any help would be appreciated if you think it can work with a node socket.
Thanks @FGasper

**Reply by FGasper** [reply]
@brandonmpetty zmodem.js doesn’t really care about the underlying transport.

It should work just to have your Sentry object’s `sender` write to your socket, send all reads from your socket through the Sentry’s `.consume()`, and have the Sentry’s `to_terminal` callback send to your BBS logic.
